src/colorizer/Dataset.ts

```diff
@@ -125,42 +148,57 @@ export default class Dataset {
     return isFeatureType(inputType) ? inputType : defaultType;
   }
 
-  /**
-   * Loads a feature from the dataset, fetching its data from the provided url.
-   * @returns A promise of an array tuple containing the feature key and its FeatureData.
-   */
-  private async loadFeature(metadata: ManifestFile["features"][number]): Promise<[string, FeatureData]> {
+  private parseManifestFeatureData(metadata: ManifestFile["features"][number]): FeatureInfo {
```

Comment:
So the original `loadFeature` method took the metadata's feature type (`ManifestFile["features"][number]`), validated its properties, and fetched the data to make it into a `FeatureData` type.

This change breaks those steps into two functions; `validateManifestFeatureData` validates it into the new `FeatureInfo` type, and `loadFeature` does the data fetching.